### DISCUSSION TITLE
동영상 촬영에서 카메라 설정이 궁금합니다.

### DISCUSSION THREAD

**Discussion by peterkangdev** [naverz/zepeto-studio-kor#1139] [5. Q&A]
혹시 동영상 촬영 예제 같은거 없나요?
제페토 가이드에 있는거 따라서 만들어 보긴 했는데
사용해 보니 저로서는 좀 어렵네요.. 혹시 아시는 분 힌트라도 알려주시면 감사하겠습니다.

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 동영상 예제에 스크립트만 있다보니 다들 막막해 하시는 것 같습니다.
가이드에도 스크린샷을 보강할 예정입니다.

우선 캔버스에서 필요한 버튼들을 생성시켜 주시고, 
https://studio.zepeto.me/guides/video-save-share-post 에 있는 코드를 똑같이 만들어 주세요.

```typescript
import { Camera, GameObject, RenderTexture } from 'UnityEngine';
import { Button, RawImage, Text } from 'UnityEngine.UI';
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
import { VideoResolutions, WorldVideoRecorder } from 'ZEPETO.World'

export default class VideoRecorder extends ZepetoScriptBehaviour {


    /* WorldVideoRecorder Video UI */
    public startRecordingButton: Button;
    public stopRecordingButton: Button;
    public saveVideoButton: Button;
    public shareVideoButton: Button;
    public createFeedButton: Button;
    public getThumbnailButton: Button;
    public thumbnailImage: RawImage;
    

    /* Recorder Camera */
    public recorderCamera: Camera;

    /* Target Texture */
    public targetTexture: RenderTexture;

    /* VideoPlayer Object */
    private videoPlayerObject: GameObject;


    Awake() {
        this.videoPlayerObject = new GameObject();
        this.recordingText.text = "";
    }

    Start() {
        this.startRecordingButton.onClick.AddListener(() => {
            if (false == WorldVideoRecorder.StartRecording(this.recorderCamera, VideoResolutions.W1280xH720, 15))
                return;
            
            this.StartCoroutine(this.CheckRecording());
        });

        this.stopRecordingButton.onClick.AddListener(() => {
            if (false == WorldVideoRecorder.IsRecording())
                return;
            
            WorldVideoRecorder.StopRecording();
        });

        this.saveVideoButton.onClick.AddListener(() => {
            if (false == WorldVideoRecorder.IsRecording()) {
                
                WorldVideoRecorder.SaveToCameraRoll((result) => { console.log(`${result}`) });

            }
        });

        this.shareVideoButton.onClick.AddListener(() => {
            if (false == WorldVideoRecorder.IsRecording()) {
                
                WorldVideoRecorder.Share((result) => { console.log(`${result}`) });
            }
        });

        this.createFeedButton.onClick.AddListener(() => {
            if (false == WorldVideoRecorder.IsRecording()) {
                
                WorldVideoRecorder.CreateFeed("[contents]", (result) => { console.log(`${result}`) });

            }
        });

        this.getThumbnailButton.onClick.AddListener(() => {
            if (false == WorldVideoRecorder.IsRecording()) {
                
                this.thumbnailImage.texture = WorldVideoRecorder.GetThumbnail();
            }
        });
    }

    *CheckRecording() {
        while (WorldVideoRecorder.IsRecording()) {
            yield null;
        }

        let videoPlayer = WorldVideoRecorder.AddVideoPlayerComponent(this.videoPlayerObject, this.targetTexture);

        if (videoPlayer == null)
            return;

        videoPlayer.Play();
    }
}
```

그 후 인스펙터에 버튼 및 썸네일 이미지, 카메라, 타겟 텍스쳐를 할당하여 주세요. 
썸네일 이미지를 받아오는 것은 UI>Raw Image 를 생성하시면 되고,
카메라는 메인 카메라를 사용했고,
타겟 텍스쳐의 경우 Create > Render Texture로 생성한 후 이름만 바꾸었습니다. 

![Screen Shot 2022-08-09 at 3 06 20 PM](https://user-images.githubusercontent.com/100118650/183583210-86d9895a-7af9-457d-8f1a-a850bb5fefc1.png)

QR 모바일 빌드를 통해 실행시키면 큰 문제없이 동작할 것입니다.

![IMG_4978](https://user-images.githubusercontent.com/100118650/183584433-8ac7c212-5e49-4031-80ca-e64095c43d2d.PNG)

이 예제를 바탕으로 좀 더 효율적으로 배치하고 응용해 보세요.
감사합니다.

**Comment by peterkangdev** [comment]
Recorder Camera 를 메인카메라를 사용하셨다고 했는데, 메인카메라는 프리팹으로 되어 있는걸 사용하란 말씀이신지?
아니면 script 에서 ZepetoPlayers.instance.LocalPlayer.zepetoCamera.camera 를 사용하란 것인지 궁금합니다. 
물론 둘 다 해봤는데 프리팹은 당연히 안되고, 
스크립트는 동영상 촬영은 되는데, 메인 화면이 나오질 않습니다 
카메라 적용을 자세히 알려주시면 고맙겠습니다.

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요, https://studio.zepeto.me/kr/guides/video-save-share-post 가이드 하단의 인스펙터 스크린샷을 참고해서 세팅했는데도 제대로 동작하지 않고 있나요?
예시에서의 카메라는 프리팹 형태가 아닌 Hierarchy에 생성된 메인카메라를 그대로 드래그해서 설정했습니다.

**Reply by peterkangdev** [reply]
저는 프리펩 메인카메라를 Hierachy 에 넣지 않고, ZepetoPlayers Editor 에 있는 Camera Prefab 란에 넣어 사용합니다.
그래서 어떻게 넣는건지 의아해서 질문드리는 겁니다. 
(혹시 일반적으로 카메라 프리펩을 Hierachy 에 넣어서 사용해야 하나요?)

**Reply by Zepeto-TechEvangelist** [reply]
가이드에 명시된 샘플은 가장 간단한 방법을 안내하기 위해 아래와 같이 제작되었습니다.
![Screen Shot 2022-09-26 at 11 10 50 AM](https://user-images.githubusercontent.com/100118650/192180444-565a92d1-839d-4733-9adc-461e8237523e.png)

일반적인 방식이 동영상 촬영이 아닌 월드 제작시 캐릭터를 비춰주는 카메라에 대한 이야기라면
Zepeto Players컴포넌트를 추가하고 캐릭터를 생성시키면 캐릭터 컨트롤러와 함께 제페토 카메라가 자동으로 생성되는 형태입니다.
따라서 별도의 카메라가 하이어라키에 존재하지 않아도 제페토 카메라로 세팅이 되어서 플레이가 되는 형태입니다.
ZepetoPlayers.instance.LocalPlayer.zepetoCamera.camera 는 제페토 카메라를 가리키는 코드입니다.

동영상 촬영의 경우에는 동영상 녹화에 사용될 카메라를 지정하는 형태이므로 샘플에 명시된 코드를 그대로 사용하실 경우
인스펙터에서 카메라를 설정하도록 코드가 만들어졌기에
안내드린 방법을 사용하시는 것이 좋습니다.
